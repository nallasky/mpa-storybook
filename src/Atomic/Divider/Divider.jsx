import React from "react";
import { css } from '@emotion/react';
import PropTypes from "prop-types";


/**
 * Divider 컴포넌트는 콘텐츠와 콘텐츠 사이를 구분할 때 사용합니다. 
 */
const Divider = ({ color, type, width, height, style }) => {
  return (
    <hr css={[
      { width: width || '100%', borderWidth: height || '1px' },
        colors[color],
        types[type],
        style
      ]}
    />
  );
};

Divider.propTypes = {
  /**
  * Divider의 색상을 설정합니다.
  */
  color: PropTypes.oneOf(['gray', 'black']),
  /** Divider의 스타일을 설정합니다. */
  type: PropTypes.oneOf(['dot', 'dash', 'solid']),
  /** Divider의 너비를 설정합니다. */
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** Divider의 높이를 설정합니다. */
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  /** Divider에 별도의 스타일을 지정하는 경우에 설정합니다. */
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array])
};

Divider.defaultProps = {
  color: 'gray',
  type: 'dash'
};

const colors = {
  'gray': css`
    border-color: rgba(0,0,0,.1);
  `,
  'black': css`
    border-color: black;
  `
};

const types = {
  'dot': css`
    border-style: dotted
  `,
  'dash': css`
    border-style: dashed
  `,
  'solid': css`
    border-style: solid
  `
};

export default Divider;