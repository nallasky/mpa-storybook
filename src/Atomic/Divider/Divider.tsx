/** @jsx jsx */
import { jsx, css } from '@emotion/core';

type DividerProps = {
  /** Divider의 색상을 설정합니다. */
  color: 'gray' | 'black';
  /** Divider의 스타일을 설정합니다. */
  type: 'dot' | 'dash' | 'solid';
  /** Divider의 너비를 설정합니다. */
  width?: string | number;
  /** Divider의 높이를 설정합니다. */
  height?: string | number;
  /** Divider에 별도의 스타일을 지정하는 경우에 설정합니다. */
  style?: {}
};

/** Divider 컴포넌트는 콘텐츠와 콘텐츠 사이를 구분할 때 사용합니다. */
const Divider = ({ color, type, width, height, style }: DividerProps) => {
  return (
      <hr css={[
        { width: width || '100%', borderWidth: height || '1px' },
          colors[color],
          types[type],
          style
        ]}
      />
  );
};

Divider.defaultProps = {
  color: 'gray',
  type: 'dash'
};

const colors = {
  'gray': css`
    border-color: rgba(0,0,0,.1);
  `,
  'black': css`
    border-color: black;
  `
};

const types = {
  'dot': css`
    border-style: dotted
  `,
  'dash': css`
    border-style: dashed
  `,
  'solid': css`
    border-style: solid
  `
};

export default Divider;