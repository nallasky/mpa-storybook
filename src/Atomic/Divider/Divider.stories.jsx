import React from "react";
import { css } from '@emotion/react';
import Divider from './Divider';
import { withTests } from "@storybook/addon-jest";
import results from '../../.jest-test-results.json';

export default {
  title: 'Atomic/Divider',
  component: Divider,
  decorators: [withTests({ results })]
};

const Template = (args) => <Divider {...args} />

export const Basic = Template.bind({});
Basic.args = {
  color: 'gray',
  type: 'dash'
};

const dividerWrapper = css`
  text-align: left;
  & > div + div {
    margin-top: 2rem;
  }
`;

export const Colors = () => {
  return (
    <div css={dividerWrapper}>
      <div>
        <Divider color="gray" />
      </div>
      <div>
        <Divider color="black" />
      </div>
    </div>
  )
};

export const Types = () => {
  return (
    <div css={dividerWrapper}>
      <div>
        <Divider type="dot" />
      </div>
      <div>
        <Divider type="dash" />
      </div>
      <div>
        <Divider type="solid" />
      </div>
    </div>
  )
};

export const Width = () => {
  return (
    <div css={dividerWrapper}>
      <div>
        <Divider width="200px" style={{ margin: 0 }} />
      </div>
      <div>
        <Divider width="50%" style={{ margin: 0 }} />
      </div>
      <div>
        <Divider width="70%" style={{ margin: 0 }} />
      </div>
      <div>
        <Divider width="50rem" style={{ margin: 0 }} />
      </div>
      <div>
        <Divider width="100%" style={{ margin: 0 }} />
      </div>
    </div>
  )
};

export const Height = () => {
  return (
    <div css={dividerWrapper}>
      <div>
        <Divider />
      </div>
      <div>
        <Divider height="3px" />
      </div>
      <div>
        <Divider height="0.5rem" />
      </div>
      <div>
        <Divider height="10px" />
      </div>
    </div>
  );
};