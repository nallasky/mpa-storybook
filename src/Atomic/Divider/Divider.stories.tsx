/** @jsx jsx */
import React from 'react';
import { withKnobs  } from '@storybook/addon-knobs';
import { jsx, css } from '@emotion/core';
import Divider from '@divider/Divider';

export default {
  title: 'Atomic|Divider',
  component: Divider,
  decorators: [withKnobs]
};

export const divider = () => <Divider />;

divider.story = {
  name: 'Default'
};

const dividerWrapper = css`
  text-align: left;
  & > div + div {
    margin-top: 2rem;
  }
`;

export const colors = () => {
  return (
    <div css={dividerWrapper}>
      <div>
        <Divider color={'gray'} />
      </div>
      <div>
        <Divider color={'black'} />
      </div>
    </div>
  )
};

export const types = () => {
  return (
      <div css={dividerWrapper}>
        <div>
          <Divider type={'dot'} />
        </div>
        <div>
          <Divider type={'dash'} />
        </div>
        <div>
          <Divider type={'solid'} />
        </div>
      </div>
  )
};

export const width = () => {
  return (
      <div css={dividerWrapper}>
        <div>
          <Divider width={"200px"} style={{ margin: 0 }} />
        </div>
        <div>
          <Divider width={'50%'} style={{ margin: 0 }} />
        </div>
        <div>
          <Divider width={'70%'} style={{ margin: 0 }} />
        </div>
        <div>
          <Divider width={'50rem'} style={{ margin: 0 }} />
        </div>
        <div>
          <Divider width={'100%'} style={{ margin: 0 }} />
        </div>
      </div>
  )
};

export const height = () => {
  return (
      <div css={dividerWrapper}>
        <div>
          <Divider />
        </div>
        <div>
          <Divider height={"3px"} />
        </div>
        <div>
          <Divider height={'0.5rem'} />
        </div>
        <div>
          <Divider height={'10px'} />
        </div>
      </div>
  );
};