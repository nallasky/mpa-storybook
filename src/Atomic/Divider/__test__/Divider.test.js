import React from 'react';
import Divider from "../Divider";
import { mount } from "enzyme";

describe("Divider test", () => {
  it("matches snapshot (initialProp)", () => {
    const wrapper = mount(<Divider />);
    expect(wrapper).toMatchSnapshot();
  });

  it ("matches snapshot (black color)", () => {
    const wrapper = mount(<Divider color="black" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (dot type)", () => {
    const wrapper = mount(<Divider type="dot" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (solid type)", () => {
    const wrapper = mount(<Divider type="solid" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (change width)", () => {
    const wrapper = mount(<Divider width="400px" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (change height)", () => {
    const wrapper = mount(<Divider height="1rem" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("matches snapshot (custom style)", () => {
    const wrapper = mount(<Divider style={{ borderColor: "white" }} />);
    expect(wrapper).toMatchSnapshot();
  });
});