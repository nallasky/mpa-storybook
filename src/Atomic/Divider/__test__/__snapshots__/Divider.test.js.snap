// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider test matches snapshot (black color) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (change height) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (change width) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (custom style) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (dot type) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (initialProp) 1`] = `ReactWrapper {}`;

exports[`Divider test matches snapshot (solid type) 1`] = `ReactWrapper {}`;
